src/librustc_borrowck/borrowck/mod.rs

```diff
@@ -746,6 +747,19 @@ impl<'a, 'tcx> BorrowckCtxt<'a, 'tcx> {
                 },
                 moved_lp.ty));
         }
+        if let (Some(CompilerDesugaringKind::ForLoop), Ok(snippet)) = (
+            move_span.compiler_desugaring_kind(),
+            self.tcx.sess.source_map().span_to_snippet(move_span),
```

Comment:
Nit: not sure whether `span_to_snippet` is expensive or not, but if it is, them it may make sense to nest `if let`s to avoid pessimizing.